dataformat-xml-dom/src/test/java/org/camunda/spin/xml/dom/format/spi/DomXmlDataFormatProtectionTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.spin.xml.dom.format.spi;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import org.camunda.spin.DataFormats;
+import org.camunda.spin.impl.xml.dom.format.DomXmlDataFormat;
+import org.camunda.spin.xml.SpinXmlDataFormatException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class DomXmlDataFormatProtectionTest {
+
+  protected static DomXmlDataFormat format;
+
+  @BeforeClass
+  public static void setUpMocks() {
+    format = (DomXmlDataFormat) DataFormats.xml();
+  }
+
+  @Test
+  public void shouldThrowExceptionForTooManyAttributes() {
+    // given
+    String testXml = "org/camunda/spin/xml/dom/format/spi/FeatureSecureProcessing.xml";
+    InputStream testXmlAsStream = this.getClass().getClassLoader().getResourceAsStream(testXml);
+
+    // when
+    assertThatThrownBy(() -> {
+      format.getReader().readInput(new InputStreamReader(testXmlAsStream));
+    })
+        // then
+        .isInstanceOf(SpinXmlDataFormatException.class);
+  }
+
+  @Test
+  public void shouldNotOverrideExternalSchemaAccess() {
+    // given
+    System.setProperty("javax.xml.accessExternalSchema", "");
+    String testXml = "org/camunda/spin/xml/dom/format/spi/ExternalSchemaAccess.xml";
+    InputStream testXmlAsStream = this.getClass().getClassLoader().getResourceAsStream(testXml);
+
+    try {
+      // when
+      format.getReader().readInput(new InputStreamReader(testXmlAsStream));
+    } finally {
+      // then
+      System.clearProperty("javax.xml.accessExternalSchema");
+    }
+  }
```

Comment:
As discussed with you, this test does not seem to work as intended. The system property is set, but `DomXmlDataFormat` is created before so the property is read before setting it in the test.

This test does not have any assertions. Do we expect an exception here like in [this test](https://github.com/camunda/camunda-bpm-platform/blob/13a31373a479abd4e5fbe47b7c1e57dabac0dba2/model-api/xml-model/src/test/java/org/camunda/bpm/model/xml/impl/parser/ParserTest.java#L51-L75)?

Reply:
I managed to set the property before the `DomXmlDataFormat` is created, but the test still passes. I wonder if it makes sense to include it at all. The fix related to the test was related to JBoss, so it doesn't make sense to include it in a standalone library like Spin.

What do you think?

Reply:
Good point, let's remove it from Spin.